src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -362,7 +363,10 @@ async def package_python_dist(
     field_set: PythonDistributionFieldSet,
     python_setup: PythonSetup,
 ) -> BuiltPackage:
-    transitive_targets = await Get(TransitiveTargets, TransitiveTargetsRequest([field_set.address]))
+    transitive_targets = await Get(
+        TransitiveTargets,
+        TransitiveTargetsRequest([field_set.address], include_special_cased_deps=True),
```

Comment:
Was `include_special_cased_deps=True` necessary? I think we had found it's not iirc.

We very likely do not want to set this because it results in some weird behaviors. For example, Pants has the `relocated_files` target type, which depends on some `files` targets. If `include_special_cased_deps=True`, then both the original `files` and `relocated_files` will be used, which is not usually wanted outside of when users are doing project introspection via `./pants dependencies` and `./pants dependees`.